client.js

```diff
@@ -72,12 +72,85 @@
 		readyCallbacks = {},
 		communicators = {};
 
+	var _getOption = function (key, defaultValue) {
+		if(!options) {
+			return defaultValue;
+		}
+		return options[key] || defaultValue;
+	};
+
 	self.fileInputElementId = "siofu_input_"+window.siofu_global.instances++;
 	self.resetFileInputs = true;
-	self.useText = false;
-	self.serializedOctets = false;
-	self.useBuffer = true;
-	self.chunkSize = 1024 * 100; // 100kb default chunk size
+	self.useText = _getOption("useText", false);
+	self.serializedOctets = _getOption("serializedOctets", false);
+	self.useBuffer = _getOption("useBuffer", true);
+	self.chunkSize = _getOption("chunkSize", 1024 * 100); // 100kb default chunk size
+	self.topicName = _getOption("topicName", "siofu");
+
+  /**
+  * Used with wrapOptions to use only one topic for all event instead of using separate topic
+  */
+	self.onlyOneTopic = _getOption("onlyOneTopic", false);
+
+
+	/**
+	 * Wrap options allow you to wrap the Siofu messages into a predefined format.
+	 * You can then easily use Siofu packages even in strongly typed topic.
+	 * WrapOptions is an object constituted of two mandatory key and one optional:
+	 * siofuDataKey (mandatory): Corresponding to the key where we will send the siofu data
+	 * siofuActionKey (mandatory): Corresponding to the key where we will send the siofu action type
```

Comment:
This seems like *too* much configuration to me.  Why do the keys need to be variable?  I don't want this feature to be unnecessarily hard to use.  Would it work for you to give the keys pre-defined names like `siofuAction` and `siofuMessage` and don't let them be configurable.

Forcing the key names to be constant also eliminates the need for `unwrapOptions`.

Reply:
To be integrated in an existing typed topic.
I have a topic that receive something like { userId, topic, message } and return { acknowledgement, data }.
I added an optional action message to adapt but i wouldn't want to impose the message and data key that are specific to my implementation.
I can propose you a default action that is configurable. Like this:
option name: wrapData
Default value: false
Possible value: true, false or object. Object structure:
```
{
  wrapKey: {
    action: 'action',
    data: 'data'
  },
  unwrapKey: {
    action: 'action',
    data: 'data'
  }
}
```
If wrapData is false, not an object or not well configured we don't use it.
If wrapData is true we use it with default key action and data.
If wrapData is objetct and well configured we use it specified keys.

What do you think ?

Reply:
That looks good to me.  Less prone to user error because all of the settings are encapsulated in one option bag.  Thanks!

`additionnalData` should be `additionalData`

Reply:
Done